flocker/node/functional/test_docker.py

```diff
@@ -170,6 +170,59 @@ def started(_):
         d.addCallback(started)
         return d
 
+    def test_private_registry_image(self):
+        """
+        ``DockerClient.add`` can start containers based on an image from a
+        private registry.
+        """
+        address_user = socket.socket()
+        self.addCleanup(address_user.close)
+        address_user.bind(('', 0))
+        used_address = address_user.getsockname()
+        address_user.close()
+        registry_name = random_name(self)
+        registry_port = used_address[1]
+        registry_starting = self.start_container(
+            unit_name=registry_name,
+            image_name='registry:2',
```

Comment:
I should probably add the registry image to the  list of images pre-pulled by the build slave.